public/pages/Rollups/containers/Rollups/Rollups.tsx

```diff
@@ -0,0 +1,583 @@
+  rollupService: RollupService;
+}
+
+interface RollupsState {
+  totalRollups: number;
+  from: number;
+  size: number;
+  search: string;
+  sortField: keyof RollupItem;
+  sortDirection: Direction;
+  selectedItems: RollupItem[];
+  rollups: RollupItem[];
+  rollupExplain: any;
+  loadingRollups: boolean;
+  isPopoverOpen: boolean;
+  isDeleteModalVisible: boolean;
+}
+
+export default class Rollups extends Component<RollupsProps, RollupsState> {
+  constructor(props: RollupsProps) {
+    super(props);
+
+    const { from, size, search, sortField, sortDirection } = getURLQueryParams(this.props.location);
+
+    this.state = {
+      totalRollups: 0,
+      from,
+      size,
+      search,
+      sortField,
+      sortDirection,
+      selectedItems: [],
+      rollups: [],
+      loadingRollups: false,
+      isPopoverOpen: false,
+      isDeleteModalVisible: false,
+      rollupExplain: {},
+    };
+
+    this.getRollups = _.debounce(this.getRollups, 500, { leading: true });
+  }
+
+  async componentDidMount() {
+    chrome.breadcrumbs.set([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.ROLLUPS]);
+    await this.getRollups();
+    const { rollups } = this.state;
+    if (rollups.length != 0) await this.getExplains();
+  }
+
+  async componentDidUpdate(prevProps: RollupsProps, prevState: RollupsState) {
+    const prevQuery = Rollups.getQueryObjectFromState(prevState);
+    const currQuery = Rollups.getQueryObjectFromState(this.state);
+    if (!_.isEqual(prevQuery, currQuery)) {
+      await this.getRollups();
+      const { rollups } = this.state;
+      if (rollups.length != 0) await this.getExplains();
+    }
+  }
+
+  static getQueryObjectFromState({ from, size, search, sortField, sortDirection }: RollupsState): RollupsQueryParams {
+    return { from, size, search, sortField, sortDirection };
+  }
+
+  getRollups = async (): Promise<void> => {
+    this.setState({ loadingRollups: true });
+    try {
+      const { rollupService, history } = this.props;
+      const queryParamsString = queryString.stringify(Rollups.getQueryObjectFromState(this.state));
+      history.replace({ ...this.props.location, search: queryParamsString });
+      const rollupJobsResponse = await rollupService.getRollups(queryParamsString);
+      if (rollupJobsResponse.ok) {
+        const { rollups, totalRollups } = rollupJobsResponse.response;
+        this.setState({ rollups, totalRollups });
+      } else {
+        toastNotifications.addDanger(rollupJobsResponse.error);
+      }
+    } catch (err) {
+      toastNotifications.addDanger(getErrorMessage(err, "There was a problem loading the rollups"));
+    }
+    this.setState({ loadingRollups: false });
+  };
+
+  getExplains = async (): Promise<void> => {
+    this.setState({ loadingRollups: true });
+    try {
+      const { rollupService } = this.props;
+      const { rollups } = this.state;
+      //Concat the rollup job ids to form params for explain api:
```

Comment:
Wouldn't rely on ordering being the same as it might not be when combining these. Use the rollupIds to guarantee you are populating the correct one. But also as noted above this needs to be moved to backend API and done in one request from browser.